File: character-counter/stack.js
class Stack{
    constructor(){
      this.items = []
    }
    push(item){
      this.items.unshift(item)
    }
    
    peek(){
      return this.items[0]
    }
    
    pop(){
      return this.items.shift()
    }
    
    isEmpty(){
      return this.items.length == 0
    }
    
    show(){
      var str = "";
      for (var i = 0; i < this.items.length; i++)
          str += this.items[i] + " ";
      return str;
    }
  }
  
  st = new Stack()
  console.log(st.isEmpty())
  st.push("First In")
  st.push("Second In")
  console.log(st.show())
  console.log(st.isEmpty())
  
  console.log(st.pop())

File: character-counter/stack.rb
class Stack
    attr_accessor :items

    def initialize
        @items = []
    end

    def push(item)
        items.unshift(item)
    end

    def peek
        return items[0]
    end

    def pop
        items.shift
    end

    def empty?
        return items.length == 0
    end

    def show
        str = ""
        for item in items do
            str += item + " "
        end
        return str
    end
end

b = Stack.new
b.push("2")
b.push("6")
b.push("-7")
b.push("5")

# b.pop
puts b.show
# b.pop
# puts b.empty?